### PR TITLE
Feature/idf image loading

### PR DESCRIPTION
**Pull request by jccrane**
This was in principle a minor change to check the series description in globbed files within svkImageReader2, however the trick was to prevent recursive calls to the method when trying to get the series description from the file list by "reading" them and in turn globbing the same file set.  So this required a bit of extra work to be able to create a reader without CanRead in the ImageReader via the factory. 

- added test case
- tests passed
- note, that I haven't changed the default behavior of applications to single file loading, but the behavior should have few side effects when globbing now.

### CODE REVIEW

File: libs/src/svkImageReader2.cc [hunk 1]
```diff
@@ -616,7 +637,33 @@ void svkImageReader2::GlobFileNames()
...
+            vtkStringArray* seriesGroup = vtkStringArray::New();  
+            for (int i = 0; i < numFilesInGroup; i++) {
+                string groupFileName = sortFileNames->GetNthGroup( groupToUse )->GetValue(i); 
+                string seriesDescription = this->GetFileSeriesDescription( groupFileName ); 
+                //cout << "SERIES DESCRIPTION: " << seriesDescription << endl;
+                if ( seriesDescription.compare( referenceSeriesDescription ) == 0 ) {
+                    seriesGroup->InsertNextValue( groupFileName ); 
+                }
+            }
+            this->SetFileNames( seriesGroup );
```
Comment (beckolson) [inline libs/src/svkImageReader2.cc]: Does seriesGroup need to be free'd?

File: libs/src/svkImageReader2.h [hunk 2]
```diff
@@ -65,20 +64,46 @@ class svkImageReader2 : public vtkImageReader2
...
+            GE_SIGNA_LX2,
+            GE_SIGNA_5X,
+            LC_MODEL_COORD, 
+            LC_MODEL_CSV, 
+            BRUKER_MRS,
+            DDF,
+            IDF,
+            LAST_TYPE = IDF
+        } ReaderType;
+
```
Comment (beckolson) [inline libs/src/svkImageReader2.h]: What is the advantage of using enums over using the class name with GetClassName?

File: libs/src/svkImageReader2.cc [hunk 3]
```diff
@@ -616,7 +637,33 @@ void svkImageReader2::GlobFileNames()
...
+      
+            vtkStringArray* seriesGroup = vtkStringArray::New();  
+            for (int i = 0; i < numFilesInGroup; i++) {
+                string groupFileName = sortFileNames->GetNthGroup( groupToUse )->GetValue(i); 
+                string seriesDescription = this->GetFileSeriesDescription( groupFileName ); 
+                //cout << "SERIES DESCRIPTION: " << seriesDescription << endl;
+                if ( seriesDescription.compare( referenceSeriesDescription ) == 0 ) {
+                    seriesGroup->InsertNextValue( groupFileName ); 
+                }
+            }
```
Comment (beckolson) [inline libs/src/svkImageReader2.cc]: Will this loop run for DICOM data as well as other formats, such as IDF's? If so will there be a big performance hit when loading DICOM series with a large number of files such as perfusion or diffusion?

**Comment by beckolson** [comment]
I have made a number of inline comments in this (mostly just to test out how they work) for this PR. I think everything looks fine and can be merged but just had a couple of minor concerns:

  1. Is the ReaderType enum necessary? It seems to me that the classname from the GetClassName would work for this.

2. Will this have a significant impact on the loading time for single-frame DICOM data with a large number of files such as perfusion data?

File: libs/src/svkImageReader2.h (hunk 2 above)
Comment (jccrane) [inline libs/src/svkImageReader2.h]: Nothing!  Just following the convention in the writers of using an enumeration.

__________________
Jason Crane, Ph.D.
Department of Radiology and Biomedical Imaging
UCSF Radiology MC 2532
Byers Hall Suite 301
CA Institute for Quantitative Biomedical Research
1700 4th St.<x-apple-data-detectors://6/0>
San Francisco, CA 94158-2330<x-apple-data-detectors://6/0>
e-mail: jason.crane@ucsf.edu<mailto:jasonc@radiology.ucsf.edu>
tel: 415.514.4426<tel:415.514.4426>
FAX: 415.514.2550<tel:415.514.2550>

On Dec 17, 2015, at 1:44 PM, Beck Olson <notifications@github.com<mailto:notifications@github.com>> wrote:


In libs/src/svkImageReader2.h<https://github.com/SIVICLab/sivic/pull/12#discussion_r47964572>:

> +            SIEMENS_RDA,
> +            VARIAN_FID,
> +            VARIAN_FDF,
> +            GE_PFILE,
> +            GE_SDBM,
> +            GE_POSTAGE_STAMP,
> +            GE_SIGNA_LX2,
> +            GE_SIGNA_5X,
> +            LC_MODEL_COORD,
> +            LC_MODEL_CSV,
> +            BRUKER_MRS,
> +            DDF,
> +            IDF,
> +            LAST_TYPE = IDF
> +        } ReaderType;
> +


What is the advantage of using enums over using the class name with GetClassName?

File: libs/src/svkImageReader2.cc (hunk 3 above)
Comment (jccrane) [inline libs/src/svkImageReader2.cc]: this function actually doesn't apply for DICOM loading.  .  I tested with a 4000 file DICOM series and it loads in the same time.   After observing that I found the block is different for DICOM anyway.

File: libs/src/svkImageReader2.cc (hunk 1 above)
Comment (jccrane) [inline libs/src/svkImageReader2.cc]: yes, thanks. done.

**Comment by jccrane** [comment]
tested performance, fixed memory leak, am going to merge this now to get ready to redeploy UCSF dev

**Pull request merged**